### DISCUSSION TITLE
blog/java/homework/recursive-streams/

### DISCUSSION THREAD

**Discussion by giscus** [MayCXC/maycxc.github.io#9] [Announcements]
# blog/java/homework/recursive-streams/

We were taught Scala in CSC110, and I really liked its functional features. Now that I am taking introductory Java courses, I need to carefully hide my affinity for FP from the OO supremacists that grade me. One of the more subtle ways I can express it is by using Java 8 functional interfaces wherever possible, which mostly means replacing all of the for loops I can with Stream operations. Today, this quest lead to me explore some new methods of recursion, and uncover some feats and shortcomings of Java 8 in the process.

https://maycxc.github.io/blog/java/homework/recursive-streams/

**Comment by MayCXC** [comment]
test comment please ignore
